tests/persistent_volumes_test.go

```diff
@@ -258,3 +266,196 @@ func TestPVCTemplateResize(t *testing.T) {
+		var agentsCount, coordinatorCount, DBServersCount = 3, 2, 3
+
+		deployment.Spec.Mode = api.NewMode(mode)
+		deployment.Spec.Environment = api.NewEnvironment(api.EnvironmentProduction)
+
+		volumeMode := corev1.PersistentVolumeFilesystem
+		deployment.Spec.DBServers.VolumeClaimTemplate = &corev1.PersistentVolumeClaim{
+			Spec: corev1.PersistentVolumeClaimSpec{
+				AccessModes: []corev1.PersistentVolumeAccessMode{
+					corev1.ReadWriteOnce,
+				},
+				VolumeMode: &volumeMode,
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceStorage: resource.MustParse("1Gi"),
+					},
+				},
+			},
+		}
+
+		deployment.Spec.DBServers.Count = util.NewInt(DBServersCount)
+		deployment.Spec.Agents.Count = util.NewInt(agentsCount)
+		deployment.Spec.Coordinators.Count = util.NewInt(coordinatorCount)
+	})
+	require.NoError(t, err)
+
+	// Create deployment
+	_, err = arangoClient.DatabaseV1alpha().ArangoDeployments(k8sNameSpace).Create(depl)
+	require.NoError(t, err, "failed to create deployment: %s", err)
+	defer deferedCleanupDeployment(arangoClient, depl.GetName(), k8sNameSpace)
+
+	depl, err = waitUntilDeployment(arangoClient, depl.GetName(), k8sNameSpace, deploymentIsReady())
+	require.NoError(t, err, fmt.Sprintf("Deployment not running in time: %s", err)) // <-- fails here at the moment
+	require.NotNil(t, depl.Spec.DBServers.VolumeClaimTemplate)
+
+	// Update deployment
+	depl.Spec.DBServers.VolumeClaimTemplate.Spec.StorageClassName = util.NewString(newStorageClassName)
+	_, err = arangoClient.DatabaseV1alpha().ArangoDeployments(k8sNameSpace).Update(depl)
+	require.NoError(t, err, "failed to update deployment: %s", err)
+
+	// Check for updated deployment
+	isStorageChanged := func(deployment *api.ArangoDeployment) error {
+		pvc := deployment.Spec.DBServers.VolumeClaimTemplate
+		if pvc == nil {
+			return fmt.Errorf("persistant volume claim can not be nil")
+		}
+		if pvc.Spec.StorageClassName == nil {
+			return fmt.Errorf("storage class name can not be nil")
+		}
+		if *pvc.Spec.StorageClassName != newStorageClassName {
+			return fmt.Errorf("storage class name has not been changed")
+		}
+
+		for _, server := range deployment.Status.Members.DBServers {
+			pvc, err := kubecli.CoreV1().PersistentVolumeClaims(k8sNameSpace).Get(server.PersistentVolumeClaimName, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if pvc.Spec.StorageClassName == nil {
+				return fmt.Errorf("storage class name can not be nil")
+			}
+			if *pvc.Spec.StorageClassName != newStorageClassName {
+				return fmt.Errorf("storage class name has not been chagned")
+			}
+		}
+		return nil
+	}
+
+	depl, err = waitUntilDeployment(arangoClient, depl.GetName(), k8sNameSpace, isStorageChanged, time.Minute*5)
+	require.NoError(t, err, "failed to change storage class for db servers: %s", err)
+
+	// Cleanup
+	removeDeployment(arangoClient, depl.GetName(), k8sNameSpace)
```

Comment:
I would love to see the following in addition:
Create a collection with 3 shards and put about 200 documents in there. Then start the test. After all volumes have been replaced, test if all the documents are still there. (They should ...)

Reply:
changed